### PR TITLE
Account: update method

### LINKED ISSUE #98: [User] Add a method to edit a user
The endpoint is on the auth service (!!not the API):

`PUT /v1/users/account`

```
    params.require(:user).permit(:username, :email, :fullname, :company, :location,
                                 :tos_accepted, :email_newsletter, :password, :current_password,
                                 :password_confirmation,)
```

And the response is the same as the one for the self() method.

### PR DESCRIPTION
**Pull request by ksol**
Fix #98

### CODE REVIEW

File: src/Users/index.js [hunk 1]
```diff
@@ -30,6 +42,24 @@ export default class Users {
...
+ * @property {?String} email
+ * @property {?String} username
+ * @property {?String} company
+ * @property {?String} location
+ * @property {?String} fullname
+ * @property {?Boolean} tos_accepted Did the user accept our TOS
+ * @property {?Boolean} email_newsletter Did the user opt in to receive newsletter
+ * @property {?String} company
+ * @property {?String} location
+ * @property {?String} fullname
```
Comment (johnsudaar) [inline src/Users/index.js]: Those fields are here twice.

File: src/Users/index.js [hunk 2]
```diff
@@ -16,6 +16,18 @@ export default class Users {
   self() {
     return unpackData(this._client.authApiClient().get('/users/self'), 'user')
   }
+
+  /**
+   * Updates the current user account informations
+   * @param {AccountUpdateParams} attributes - User attributes to update
+   * @return {Promise<User | APIError>} Promise that when resolve return the user account informations
+   */
+  updateAccount(attributes) {
```
Comment (johnsudaar) [inline src/Users/index.js]: What about juste update ?

```
client.User.update({company: 'Scalingo'}) 
```
?

Comment (ksol) [reply src/Users/index.js]: I think `update` does not convey that we're updating the current user, it's implicit. `updateAccount` does a (slightly) better job at that; `updateSelf` would be even better (but then, wouldn't reflect the endpoint)

Comment (johnsudaar) [reply src/Users/index.js]: OK !

**Pull request merged**